consensus/src/chained_bft/event_processor.rs

```diff
@@ -677,75 +508,257 @@ impl<T: Payload> EventProcessor<T> {
         };
         None
     }
+}
 
-    /// Upon (potentially) new commit:
-    /// 0. Verify that this commit is newer than the current root.
-    /// 1. Notify state computer with the finality proof.
-    /// 2. After the state is finalized, update the txn manager with the status of the committed
-    /// transactions.
-    /// 3. Prune the tree.
-    async fn process_commit(
-        &self,
-        committed_block: Arc<Block<T>>,
-        finality_proof: LedgerInfoWithSignatures<Ed25519Signature>,
-    ) {
-        // First make sure that this commit is new.
-        if committed_block.round() <= self.block_store.root().round() {
+impl<T: Payload> EventProcessor<T> {
+    pub fn new(
```

Comment:
Why did you move some of the functions to a separate `impl` block? Not that I have something against it, just trying to understand the changes (and it makes the review process more complicated because the UI shows it as new code so it's hard to see the real differences).

Reply:
I revert this change to make this PR cleaner, I'll create a separate PR for re-organizing.